pkg/env/dev.go

```diff
@@ -62,7 +62,19 @@ type dev struct {
 }
 
 func newDev(ctx context.Context, log *logrus.Entry, instancemetadata instancemetadata.InstanceMetadata) (*dev, error) {
+	// Apply dev variable overrides for RP process
+	err := os.Setenv("AZURE_CLIENT_ID", os.Getenv("AZURE_RP_CLIENT_ID"))
+	if err != nil {
+		return nil, err
+	}
+	err = os.Setenv("AZURE_CLIENT_SECRET", os.Getenv("AZURE_RP_CLIENT_SECRET"))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
don't do this - use NewClientCredentialsConfig etc. instead of NewAuthorizerFromEnvironment below.